hometask1/foo.rb

```diff
@@ -0,0 +1,46 @@
+class Foo
+  @@counter = 0
+  @instance_counter = 0
+
+  def initialize
+    self.class.counter += 1
```

Comment:
You can manage counter here as a class variable `@@counter`. It belongs not to some specific object, but to class `Foo` in general. You don't need an accessor for it.